internal/providers/terraform/azure/container_registry.go

```diff
@@ -0,0 +1,181 @@
+	}
+
+	if d.Get("georeplication_locations").Exists() {
+		nambersOflocations = len(d.Get("georeplication_locations").Array())
+	}
+	if d.Get("georeplications").Exists() {
+		nambersOflocations = len(d.Get("georeplications.0.location").Array())
+	}
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	if d.Get("georeplication_locations").Exists() || d.Get("georeplications").Exists() {
+		if nambersOflocations == 1 {
+			costComponents = append(costComponents, ContainerRegistryGeolocationCostComponent(fmt.Sprintf("Geo replication (%s location)", strconv.Itoa(nambersOflocations)), location, sku))
+		}
+		if nambersOflocations > 1 {
+			costComponents = append(costComponents, ContainerRegistryGeolocationCostComponent(fmt.Sprintf("Geo replication (%s locations)", strconv.Itoa(nambersOflocations)), location, sku))
+		}
+	}
+
+	if d.Get("georeplication_locations").Type == gjson.Null && d.Get("georeplications").Type == gjson.Null {
+		costComponents = append(costComponents, ContainerRegistryCostComponent(fmt.Sprintf("Registry usage (%s)", sku), location, sku))
+	}
+
+	if u != nil && u.Get("storage_gb").Type != gjson.Null {
+		storageGb = decimalPtr(decimal.NewFromFloat(u.Get("storage_gb").Float()))
+		if storageGb.GreaterThan(*includedStorage) {
+			overStorage = storageGb.Sub(*includedStorage)
+			storageGb = &overStorage
+			costComponents = append(costComponents, ContainerRegistryStorageCostComponent(fmt.Sprintf("Storage (over %s GB)", includedStorage), location, sku, storageGb))
+		}
+	}
+
+	if u != nil && u.Get("monthly_build_vcpu_hrs").Type != gjson.Null {
+		monthlyBuildVcpu = decimalPtr(decimal.NewFromFloat(u.Get("monthly_build_vcpu_hrs").Float() * 3600))
+	}
+
+	costComponents = append(costComponents, ContainerRegistryCPUCostComponent("Build vCPU", location, sku, monthlyBuildVcpu))
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func ContainerRegistryCostComponent(name, location, sku string) *schema.CostComponent {
+
+	return &schema.CostComponent{
+
+		Name:            name,
+		Unit:            "days",
+		UnitMultiplier:  1,
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(30)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("Container Registry"),
+			ProductFamily: strPtr("Containers"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("Container Registry")},
+				{Key: "skuName", Value: strPtr(sku)},
+				{Key: "meterName", Value: strPtr(fmt.Sprintf("%s Registry Unit", sku))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+func ContainerRegistryGeolocationCostComponent(name, location, sku string) *schema.CostComponent {
+
+	return &schema.CostComponent{
+
+		Name:            name,
+		Unit:            "days",
+		UnitMultiplier:  1,
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(30)),
```

Comment:
The Azure pricing page mentions "1.667 per replicated region" so we'll need to multiply this by the number of location